src/providers/details/tabs/Hosts/components/VSphereHostsCells.tsx

```diff
@@ -0,0 +1,57 @@
+import type { FC } from 'react';
+import type { RowProps } from 'src/components/common/TableView/types';
+import { NameCellRenderer } from 'src/modules/Providers/views/details/tabs/Hosts/components/NameCellRenderer';
+
+import type { ResourceField } from '@components/common/utils/types';
+import { Td } from '@patternfly/react-table';
+
+import type { HostCellProps, InventoryHostNetworkTriple } from './utils/types';
+import IDCellRenderer from './IDCellRenderer';
+import LinkSpeedCellRenderer from './LinkSpeedCellRenderer';
+import MTUCellRenderer from './MTUCellRenderer';
+import NetworkCellRenderer from './NetworkCellRenderer';
+
+type RenderTdProps = {
+  resourceData: InventoryHostNetworkTriple;
+  resourceFieldId: string;
+  resourceFields: ResourceField[];
+};
+
+const cellRenderers: Record<string, FC<HostCellProps>> = {
+  id: IDCellRenderer,
+  linkSpeed: LinkSpeedCellRenderer,
+  mtu: MTUCellRenderer,
+  name: NameCellRenderer,
+  network: NetworkCellRenderer,
+};
+
+/**
+ * Function to render a table cell (Td).
+ * If the cell is an inventory cell (NETWORK_COUNT, STORAGE_COUNT, VM_COUNT, or HOST_COUNT)
+ * and there's no inventory data, it won't render the cell.
+ */
+const renderTd = ({ resourceData, resourceFieldId, resourceFields }: RenderTdProps) => {
+  if (!resourceFieldId) return <Td></Td>;
+
+  const CellRenderer = cellRenderers?.[resourceFieldId] ?? (() => <></>);
+  return (
+    <Td key={resourceFieldId} dataLabel={resourceFieldId}>
+      <CellRenderer data={resourceData} fieldId={resourceFieldId} fields={resourceFields} />
+    </Td>
+  );
+};
+
+const VSphereHostsCells: FC<RowProps<InventoryHostNetworkTriple>> = ({
+  resourceData,
+  resourceFields,
+}) => {
+  return (
+    <>
+      {resourceFields?.map(({ resourceFieldId }) =>
+        renderTd({ resourceData, resourceFieldId, resourceFields }),
```

Comment:
u are using the component wrong,
<RenderTd .... />

Reply:
RenderTd is a function and not a comp...

Reply:
1. component = function
2. it isnt this? src/providers/details/tabs/Hosts/components/utils/renderTd.tsx 
https://github.com/kubev2v/forklift-console-plugin/pull/1616/files#diff-2505a8b7f489d5190854d77f7c5606947bdf118aa88d96155a0169b417fbad32R33